pkgs/swiftgen/swift2objc/lib/src/parser/parsers/declaration_parsers/class_parser.dart

```diff
@@ -0,0 +1,15 @@
+import '../../../ast/declarations/compounds/class_declaration.dart';
+import '../../_core/utils.dart';
+import '../declaration_parser.dart';
+
+class ClassParser extends DeclarationParser {
```

Comment:
The inheritance isn't really buying you anything here, especially once you remove the `parsedSymbol` getter. In fact this is another class that could be a function.

`extends` style inheritance makes it more difficult to read classes and figure out what's going on. You have to switch back and forth between the class and all its super classes to find a method and figure out which version is being invoked. It's a powerful tool but comes with a cost, so I only use it when it's really needed.

Reply:
My thought here was to enforce the signature of the parsing method. It's not a problem anymore now since we turned everything into functions as we discussed.